pkg/kubectl/resource/result.go

```diff
@@ -200,3 +201,67 @@ func (r *Result) Watch(resourceVersion string) (watch.Interface, error) {
 	}
 	return w.Watch(resourceVersion)
 }
+
+// AsVersionedObject converts a list of infos into a single object - either a List containing
+// the objects as children, or if only a single Object is present, as that object. The provided
+// version will be preferred as the conversion target, but the Object's mapping version will be
+// used if that version is not present.
+func AsVersionedObject(infos []*Info, version string) (runtime.Object, error) {
+	objects := []runtime.Object{}
+	for _, info := range infos {
+		if info.Object == nil {
+			continue
+		}
+
+		// objects that are not part of api.Scheme must be converted to JSON
+		// TODO: convert to map[string]interface{}, attach to runtime.Unknown?
+		if len(version) > 0 {
+			if _, _, err := api.Scheme.ObjectVersionAndKind(info.Object); runtime.IsNotRegisteredError(err) {
+				// TODO: ideally this would encode to version, but we don't expose multiple codecs here.
+				data, err := info.Mapping.Codec.Encode(info.Object)
+				if err != nil {
+					return nil, err
+				}
+				objects = append(objects, &runtime.Unknown{RawJSON: data})
+				continue
+			}
+		}
+
+		converted, err := tryConvert(info.Mapping.ObjectConvertor, info.Object, version, info.Mapping.APIVersion)
+		if err != nil {
+			return nil, err
+		}
+		objects = append(objects, converted)
+	}
+
+	var object runtime.Object
+	if len(objects) == 1 {
+		object = objects[0]
+	} else {
+		object = &api.List{Items: objects}
+		converted, err := tryConvert(api.Scheme, object, version, latest.Version)
+		if err != nil {
```

Comment:
No, because we converted any outside the scheme into json (above) and any remaining objects are in the scheme.  The default mixed object scenario will ignore objects already in a version (if you go internal to v1beta3, an object in v1beta1 won't be converted).

> On Apr 17, 2015, at 7:35 PM, Jeff Lowdermilk <notifications@github.com> wrote:
> 
> In pkg/kubectl/resource/result.go:
> 
> > +		}
> > +
> > +		converted, err := tryConvert(info.Mapping.ObjectConvertor, info.Object, version, info.Mapping.APIVersion)
> > +		if err != nil {
> > +			return nil, err
> > +		}
> > +		objects = append(objects, converted)
> > +	}
> > +
> > +	var object runtime.Object
> > +	if len(objects) == 1 {
> > +		object = objects[0]
> > +	} else {
> > +		object = &api.List{Items: objects}
> > +		converted, err := tryConvert(api.Scheme, object, version, latest.Version)
> > +		if err != nil {
> It looks like this function allowed to return a list of disparate objects. If that's true, wouldn't the conversion fail?
> 
> —
> Reply to this email directly or view it on GitHub.
>